staging/src/k8s.io/apimachinery/pkg/runtime/serializer/yaml/meta_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type expectedGVK struct {
+	group   string
+	version string
+	kind    string
+}
+
+func TestInterpret(t *testing.T) {
+	testCases := []struct{
+		name string
+		input string
+		expected *schema.GroupVersionKind
+		errFn func(error) bool
+	}{
+		{
+			name: "successfully interpret VK",
+			input: `apiVersion: v1
+kind: Service`,
+			expected: &schema.GroupVersionKind{"", "v1", "Service"},
+		},
+		{
+			name:     "successfully interpret GVK",
+			input: `apiVersion: core/v2
+kind: Deployment`,
+			expected: &schema.GroupVersionKind{"core", "v2", "Deployment"},
+		},
+		{
+			name:     "successfully interpret V",
+			input: `apiVersion: v1`,
+			expected: &schema.GroupVersionKind{"", "v1", ""},
+		},
+		{
+			name:     "successfully interpret K",
+			input: `kind: Service`,
+			expected: &schema.GroupVersionKind{"", "", "Service"},
+		},
+		{
+			name:     "successfully interpret empty string",
+			input: ``,
+			expected: &schema.GroupVersionKind{},
+		},
+		{
+			name:     "successfully interpret empty YAML",
+			input: `---`,
+			expected: &schema.GroupVersionKind{},
+		},
+		{
+			name:     "successfully interpret multi-doc",
+			input: `---
+apiVersion: v1
+kind: Service
+---
+apiVersion: v2
+kind: Deployment`,
+			expected: &schema.GroupVersionKind{Version: "v1", Kind: "Service"},
+		},
+		{
+			name:     "successfully interpret if only G is provided",
+			input: `apiVersion: core/`,
+			expected: &schema.GroupVersionKind{Group: "core"},
+		},
+		{
+			name:     "fail interpret if wrong syntax",
+			input: `foo`,
+			errFn: func(err error) bool {
+				return err != nil
+			},
+		},
+	// 	TODO: Add JSON tests
+	}
+
+	for _, test := range testCases {
+		actual, err := DefaultMetaFactory.Interpret([]byte(test.input))
+		switch {
+		case test.errFn != nil:
+			if !test.errFn(err) {
+				t.Errorf("unexpected error returned with test '%s', got: %v", test.name, err)
```

Comment:
'%s' -> %q

Reply:
I included the name of the test as the actual test name in `t.Run()`, so I removed the print of the name here.